src/data/fintype/basic.lean

```diff
@@ -210,15 +210,29 @@ def equiv_fin_of_forall_mem_list {α} [decidable_eq α]
 /-- There is (computably) a bijection between `α` and `fin n` where
   `n = card α`. Since it is not unique, and depends on which permutation
   of the universe list is used, the bijection is wrapped in `trunc` to
-  preserve computability.  -/
-def equiv_fin (α) [decidable_eq α] [fintype α] : trunc (α ≃ fin (card α)) :=
+  preserve computability.
+
+See `fintype.equiv_fin` for the noncomputable version,
+and `fintype.trunc_equiv_fin_of_card_eq` and `fintype.equiv_fin_of_card_eq`
+for an equiv `α ≃ fin n` given `fintype.card α = n`.
+-/
+def trunc_equiv_fin (α) [decidable_eq α] [fintype α] : trunc (α ≃ fin (card α)) :=
 by unfold card finset.card; exact
 quot.rec_on_subsingleton (@univ α _).1
   (λ l (h : ∀ x:α, x ∈ l) (nd : l.nodup), trunc.mk (equiv_fin_of_forall_mem_list h nd))
   mem_univ_val univ.2
 
+/-- There is a (noncomputable) bijection between `α` and `fin n` where `n = card α`.
+
+See `fintype.trunc_equiv_fin` for the computable version,
+and `fintype.trunc_equiv_fin_of_card_eq` and `fintype.equiv_fin_of_card_eq`
+for an equiv `α ≃ fin n` given `fintype.card α = n`.
+-/
+noncomputable def equiv_fin (α) [fintype α] : α ≃ fin (card α) :=
+by { letI := classical.dec_eq α, exact (trunc_equiv_fin α).out }
+
 theorem exists_equiv_fin (α) [fintype α] : ∃ n, nonempty (α ≃ fin n) :=
-by haveI := classical.dec_eq α; exact ⟨card α, (equiv_fin α).nonempty⟩
+by haveI := classical.dec_eq α; exact ⟨card α, (trunc_equiv_fin α).nonempty⟩
```

Comment:
Does anything actually use this lemma? It seems a little silly to say "there exists an n such that" when we obviously know `n` is `fintype.card`.